### PR TITLE
* FIX [quic] Fix the stack use after free.

### PR DESCRIPTION
**Pull request by wanghaEMQ**

### CODE REVIEW

**Review by lee-emqx — APPROVED** [review]

**Review by OdyWayne — APPROVED** [review]

File: src/mqtt/transport/quic/mqtt_quic.c [hunk 1]
```diff
@@ -175,12 +176,12 @@ mqtt_quic_pipe_timer_cb(void *arg)
 	nni_mtx_lock(&p->mtx);
 	if (!p->busy && !nni_aio_busy(p->qsaio)) {
 		// send pingreq
-		buf[0] = 0xC0;
-		buf[1] = 0x00;
+		pingbuf[0] = 0xC0;
+		pingbuf[1] = 0x00;
```
Comment (RanMaoyi) [inline src/mqtt/transport/quic/mqtt_quic.c:180]: if pingbuf always equals {0xC0, 0x00}, Maybe we can just initialize the variables in the first place?

**Review by RanMaoyi — APPROVED** [review]

File: src/mqtt/transport/quic/mqtt_quic.c (hunk 1 above)
Comment (JaylinYu) [reply src/mqtt/transport/quic/mqtt_quic.c:180]: correct

File: src/supplemental/quic/msquic_dial.c [hunk 2]
```diff
@@ -362,7 +361,7 @@ nni_quic_dialer_close(void *arg)
 				c->dial_aio = NULL;
 				nni_aio_set_prov_data(aio, NULL);
 				nng_stream_close(&c->stream);
-				nng_stream_free(&c->stream);
+				// nng_stream_free(&c->stream);
```
Comment (wanghaEMQ) [inline src/supplemental/quic/msquic_dial.c:364]: Why? These quic streams are pending. And not available for transport layer....

File: src/supplemental/quic/msquic_dial.c [hunk 3]
```diff
@@ -467,24 +466,25 @@ static void
...
 };
 static void
-quic_free(void *arg)
+quic_stream_free(void *arg)
 {
 	nni_quic_conn *c = arg;
-	nni_reap(&quic_reap_list, c);
+	NNI_FREE_STRUCT(c);
+	// nni_reap(&quic_reap_list, c);
 }
```
Comment (wanghaEMQ) [inline src/supplemental/quic/msquic_dial.c:488]: Did you forget to run nni_msquic_quic_dialer_rele?

Comment (wanghaEMQ) [reply src/supplemental/quic/msquic_dial.c:488]: Or pipe fini

**Pull request merged**